src/metrics/matcher/namespaces.go

```diff
@@ -284,25 +285,36 @@ func (n *namespaces) process(value interface{}) error {
 				shouldWatch = false
 			}
 		}
+
 		if !shouldWatch {
 			n.metrics.unwatched.Inc(1)
 			ruleSet.Unwatch()
 		} else {
 			n.metrics.watched.Inc(1)
-			if err := ruleSet.Watch(); err != nil {
-				n.metrics.watchErrors.Inc(1)
-				n.log.WithFields(
-					log.NewField("ruleSetKey", ruleSet.Key()),
-					log.NewErrField(err),
-				).Error("failed to watch ruleset updates")
-			}
+
+			watchWg.Add(1)
+			go func() {
+				// Start the watches in background goroutines so that if the store is unavailable they timeout
+				// (approximately) in unison. This prevents the timeouts from stacking on top of each
+				// other when the store is unavailable and causing a delay of timeout_duration * num_rules.
```

Comment:
What's the impact of the delay?

Reply:
@cw9 Puppet restarted all the collectors and R2 was misconfigured so all the collectors got stuck for 10 minutes waiting on these initializations because the keys were missing